models/plugin.go

```diff
@@ -516,31 +480,33 @@ func (s *registeredPlugins) add(baseMod BasePlugin, drMod DependencyResolverMod,
 		return errors.New("plugin version cannot be empty")
 	}
 
+	for _, mod := range info.PluginMods {
+		switch mod {
+		case ModTypeYaml:
+		case ModTypeDependencyResolver:
+		default:
+			return ErrUnsupportedPlugin
+		}
+	}
+
 	switch info.PluginType {
 	case PluginTypeTask:
 	case PluginTypeHook:
 	default:
 		return ErrUnsupportedPlugin
 	}
 
-	isCandidatePluginYaml := yamlMod != nil
```

Comment:
this totally removes check for duplicate plugins

Reply:
refer https://github.com/odpf/optimus/pull/567/files#diff-a252f8221d35378ca0f6616ad8512f4bd5ee6ef60e724d1399cd7dfe1ae852f7L545

Reply:
adding alternative logic to achieve same result, please check